File: PGS/Lv.0/옹알이 (1).cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<string> babbling) {
	int answer = 0;

	for (auto b : babbling) {
		int len = 0;

		if (b.find("aya") != string::npos) {
			len += 3;
		}
		if (b.find("ye") != string::npos) {
			len += 2;
		}
		if (b.find("woo") != string::npos) {
			len += 3;
		}
		if (b.find("ma") != string::npos) {
			len += 2;
		}

		if (b.size() == len) {
			answer++;
		}
	}

	return answer;
}

File: PGS/Lv.0/JAVA/옹알이 (1).java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        
        for(String b : babbling) {
            boolean possible = true;
            while(!b.isEmpty()) {
                if(b.startsWith("aya") || b.startsWith("woo")) {
                    b = b.substring(3);
                }
                else if(b.startsWith("ye") || b.startsWith("ma")) {
                    b = b.substring(2);
                }
                else {
                    possible = false;
                    break;
                }
            }
            if(possible) answer++;
        }
        
        return answer;
    }
}